src/transformers/trainer_tf.py

```diff
@@ -269,44 +327,40 @@ def train(self) -> None:
         logger.info("  Num Epochs = %d", epochs)
         logger.info("  Total optimization steps = %d", self.train_steps)
 
-        for epoch in range(start_epoch, int(epochs + 1)):
-            for training_loss in self._training_steps(train_ds, optimizer):
-                step = iterations.numpy()
-
-                if self.args.debug:
-                    with self.tb_writer.as_default():
-                        tf.summary.scalar("loss", training_loss, step=step)
+        self.epoch_logging = 0
```

Comment:
This could also be added to `__init__` along with `self.global_step`.
Note that if that's the case, it means we will log an epoch of `0` when doing evaluation with no training (see previous comment).

Reply:
Yes, indeed. If you removes the `_log` call in the eval method there is no more issue.

Reply:
Exactly, but without to make it appear in the logs when running eval/predict mode.

Reply:
We want `evaluate` to always log metrics, wherever it's called from (training loop or outside in evaluation only from example scripts).

If `self.global_step` in in `__init__`, it is harder to differentiate whether it's being called from training loop or outside of it.

We could check if `self.epoch_logging == 0` but it would not be correct because we could want to do the following:
* initialize `Trainer` with a pre-trained model
* call `evaluate` to log evaluation metrics for reference (at this point `self.epoch_logging = 0`)
* fine-tune through training loop (where `epoch_logging` will slowly increase)
* call `evaluate` to log final evaluation metrics

In order to see my evaluation metrics on a same graph with `epoch` as x-axis, I would need to have it logged at each step.
It seems a more universal solution to just log `epoch` all the time but let me know if you have any other idea.

Reply:
The role of the `_logs` method is just to log the content of `logs` dict and must be independent of how it is used. Step and epoch should not be taken into account, they both have a value during the training and they both equals 0 during evaluation.

The trainer has to run an evaluation at each `% eval_step` and output logs at each `% logging_step`. If someone wants to have an evaluation at the end of the training, add the `--do_eval` parameter will do the job.